### PR TITLE
Add circuit generator

### PR DESCRIPTION
**Pull request by vo-nil**
This PR adds circgen utility. It can be used to generate circuits of arbitrary lengths - useful for debugging.

### CODE REVIEW

File: debug-tools/debug-tools.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  ninja,
+  pkg-config,
+  cmake,
+  crypto3,
+  boost,
+  gdb,
+  cmake_modules,
+  enableDebugging,
+  enableDebug ? false,
+  runTests ? false,
```
Comment (AndreyMlashkin) [inline debug-tools/debug-tools.nix:12]: we don't need this flag, as it does not have tests

File: debug-tools/debug-tools.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  nativeBuildInputs = [ cmake ninja pkg-config ] ++ (lib.optional (!stdenv.isDarwin) gdb);
+
+  # enableDebugging will keep debug symbols in boost
+  propagatedBuildInputs = [ (if enableDebug then (enableDebugging boost) else boost) ];
+
+  buildInputs = [cmake_modules crypto3 ];
+
+  cmakeFlags =
+    [
+      "-DCMAKE_INSTALL_PREFIX=${placeholder "out"}"
```
Comment (AndreyMlashkin) [inline debug-tools/debug-tools.nix:30]: remove DCMAKE_INSTALL_PREFIX here

**Review by x-mass — COMMENTED** [review]
You must be hating Boost.Program_options :D

File: debug-tools/debug-tools.nix (hunk 1 above)
Comment (vo-nil) [reply debug-tools/debug-tools.nix]: removed three flags. `cmake_modules` must be preserved

File: debug-tools/debug-tools.nix (hunk 2 above)
Comment (vo-nil) [reply debug-tools/debug-tools.nix]: Removed

**Review by AndreyMlashkin — APPROVED** [review]

**Comment by martun** [comment]
> You must be hating Boost.Program_options :D

Yes, we should change this.

**Comment by vo-nil** [comment]
Reworked on boost::program_options

**Review by martun — APPROVED** [review]

**Pull request merged**